IMC_DataScience/IMC_OOPTutorial/blog.md

```diff
@@ -47,25 +49,34 @@ To start, let’s make a public class called “Person” (since that’s the ty
 
 ![img](https://lh6.googleusercontent.com/zehLJy0dg7jY0SIAesHgX1_GRqjn_et0uq-QoJPPNZWSta4vbNkB1AlJnqcahxzgZHD-pG96n0ZiGzIKNM0W2T6cbDEwZ256ye3eybvMx-EX5dhS1-eXaI5Yuc48F4uLhsRLb-zf)
 
-To clarify, we’re making this object in its own file. To actually create a “person” object, we’ll just leave the “Person.java” file in the same directory as whatever program is actually using the object, then create an object in that file. How do we actually create the object, though? Well, that uses something we call a **constructor**. A constructor is a bit of code in the object class (the Person.java file) that lets you know how exactly you’re going to create a Person object in your code. Here’s a sample one I wrote:
 
-![img](https://lh6.googleusercontent.com/-2fB1lh2VpfFGMAo1ZkOUvsuq_2WVJI5olcYobpgnLRUZOcYg1KmihudAuU7cnUsgmudentdtre8LAL4KL5tdC2kJW5a-sq_Bd15Kio6J8dDgcpvfzTBwYe0bl-ZqUMab1Wuw3xT)
+
+To clarify, we’re making this object in its own file. To actually create a “person” object, we’ll just leave the “Person.java” file in the same directory as whatever program is actually using the object, then create an object in that file. How do we actually create the object, though? Well, that uses something we call a **constructor**. A constructor is a bit of code in the object class (the Person.java file) that lets you know how exactly you’re going to create a Person object in your code. Here are two that I wrote:
+
+![image-20200726115504153](C:\Users\MMO\AppData\Roaming\Typora\typora-user-images\image-20200726115504153.png)
+
+
 
 So, what’s going on here?
-Whenever we make an object, we’re making a copy of this Person object with its own firstName, lastName, and age values. These lines of code mean that, when we want to make a Person in our code, we need to give it a first name, a last name, and an age - like this!
+Whenever we make an object, we’re making a copy of this Person object with its own firstName, lastName, and (optionally) age values. These lines of code mean that, when we want to make a Person in our code, we need to give it a first name, a last name, and, if we want, an age - like this!
 
 
 
 ![img](https://lh3.googleusercontent.com/PI2YJqN66b9MmyfB4FK_BWVjicH50ctnGYhrqyaU5ripJs896i9XJDYz1ScvW-s7gH52A_PFcI66GBaxY-zlWCNim7mgyIbMmWuQFRs2QWtNTs__3YlMOcI8Akrt8yvpHN4IL0tn)
 
 
 
-Here’s an object in action. We’re making a Person object named “John” with a first name input of “John” (shocker), a last name input of “Smith”, and an age input of 26. Note that this is in a completely separate file. We’re just storing Person.java and PersonTest.java in the same directory. 
```

Comment:
Something like this side comment (shocker) is another example of that stream of consciousness style of writing